# Olfactory-Bitmap
Anticipatory File Structure for Massively Parallel, Anthropoid SAW and / or FOIC Chemical Sensors

The idea is not to map intensity values, collected in analog and converted and stored digitally, and store the respective spatial orientations of each of those scalar quantities, and thusly reproduce the percieved optical gamut, in a mosaic pattern called an "image", of the average H. sapiens (as is standard in the normal image bitmap with which you are presently familiar) but instead, for olfaction, a "look-up" table, similar to the image bitmap "color table" (for BMPs less than 24-bit RGB), is considered. This could be empolyed in a "Go-NoGo" sensor, binary, or in a sensor that is calibrated with separate sensitivities and can store with significant bit depth, according to each specific technology, or ligands, employed.

The "gamma-ut" (see "chanson' and the "monks of old"), it is presumed, will closely follow recent genomic and proteomic work on the Olfactory Receptor Genes (ORGs) and Olfactory Receptor Proteins (ORPs) of the "average human being". Despite the fact that science prersently lacks a viable or uncontested theory to explain the gamut of this sensory modality, it would behoove researchers to organize their research activities along certain lines of enquiry. LadnarofLadner (Randall Meyer) has identified at least two promising technologies, though more certainly exist. The first, probably better studied, would be the Surface Acoustic Wave (SAW) Chemical Sensor (See White and Voltmer, 1965; Wohltjen, 1979). Chief difficulties in implmenting this technology, as with all candidate technologies, will be finding appropriate ligands to mimic the approximately 400-450 ORGs/ORPs employed by the average H. sapiens. Furthermore, olfaction is a complicated process in biological organisms, involving slime and mucus and various other unecessary complications for electronic sensor transducer design. In particular, Olfactory Binding Proteins (OBPs) in the olfactory mucosa serve the initial role of attaching to the generally hydrophobic aroma molecules and bringing them into the aqueous environment that the olfactory neurons inhabit. It is then, here, that the ORGs/ORPs can act upon the aroma molecules.

This present project has, thus far, produced a 'lookup table" based upon the Yale/ TexasTech ORDB gene numbering scheme, and, in general, based upon synonymous entries (namely, Consensus Label, HORDE, Senomyx, EBI, ORDB, and Nei Lab.) provided in the table ( here : https://senselab.med.yale.edu/OrDB/info/humanorseqanal ), on their website, version 6 ( see here : https://senselab.med.yale.edu/OrDB/info/ordb_versions ), as download by LadnarofLadner circa February/ March of 2023. In particular, separate tables for the Consensus Label and for the ORDB "ORL #'s" have been devised, but tables for the other names will follow, perfunctorily, at my leisure. The tables will map the genes, proteins, and putative gene positions, and where available, implied functions of said ORGs/ORPs, to several private areas, and restricted areas, of the Unicode Planes, utilizing approximately 1800 to 2000 unique code points which will stand for ANALOGOUS receptor molecules, as utilized by SAW and FOIC transducer researchers who wish to embark upon a program of Massively Parallel Anthropoid Olfactory and Gustatory Transducers (MPAOGT : SAW or FOIC or "other"). It is not lost on the author that this usage of unicode is novel, but perhaps not without precedent, as "emoticons" are now covered by the Unicode. The truly novel element here is that the code points claimed here will not have any glyph assigned to them, but instead will represent an ineffable and presently indescribable, heretofore undigitizeable quanitity or quality; namely, aroma. The code points claimed will probably span the region of approx. U+ E000 to about E7FF, and in the two higher planes, 15 and 16, U+ EE000 to approx. EE7FF, and also U+ 10E000 to approx. 10E7FF. As for the restricted space I intend to claim, I think it will probably be U+ 3E000 to approx. 3E7FF. The similarity of the regions utilized will allow for efficeint and conveneint alteration of data, should it be required to change from a file utilizing one standard, into a file utilizing another standard. Its should also be mentioned that there is no patent or copyright on hexadecimmal codes, so, they are presumably free to use in any way, by any person. It just so happens that Unicode is a convenient and widely adhered to standard that happens to have nearly innumerable available code points to play with. 

So, in summary of this founding "README" file (and now README "2.0"; added Jan. 2024) I will remind you of the Image Bitmap sections: (1) Header, (2) Info Header, (3) Color Table, (4) Pixel Data. In analogy, the "BM" ASCII text indicating to application programs that the file is indeed a bitmap image, will instead be "BOBM" standing for, jocularly, "B.O. Bitmap" or, in a more serious vein, "Basic Olfactory Bit Map", in ASCII, and programs designed to access Olfactory Bitmaps should look for this to be prominently placed, in the first position of the Header, at Byte one ( and two?).

Presently, I have no analogy for the Info Header, but presumably, such a standard will evolve in the coming months.

Presently, the "lookup table", unicode code points are stationary and ordered, being, with some degree of imprecision and subject to some revision, more or less a stable pattern existing in nature ( i.e. the order of the discovered genes within the human genome determines the proper position of the analogous receptor-chemicals--and their resulting data---as designed and utilized by SAW and FOIC and other reserchers.), should not change over time, and need not be included within the file. It can be assumed; both the order of the ORGs and ORPs and of their analogs on the electronic transducers and / or such data as collected by said tranducers. This takes care of the analogy to the "Color Table", as this section is obviated, and unecessary.

Finally, the "pixel data"--or rather, "olfixel" data--can be as little as 1 bit per "code-point-address" (for a "Go-NoGo" sensor), but if various researchers are taking magnitude data and wish to keep it in the array, this "Pixel Data" section of the Image Bitmap will have its analog in the "Olfixel Data" section of an Olfactory Bitmap. Perhaps, as Section 2 , the analog to the "Info Header' takes shape, more details regarding how to store magnitude data and how to store "file type" data ( i.e. for Go-NoGo, binary type sensors, vs a file type which wishes to store magnitude data), will be developed. Today is, quite accidentally, April Fool's Day 2023 ("2.0" typo fix and revision and extension ; January 25, 2024). I think, in the latter case, magnitudes can be stored using code points of the analog chemicals as "addresses", and stored at that address a magnitude of any (reasonable) pre-determined size, specified in the "Info Header" of an Olfactory Bitmap.

But I havent really thought it through very thoroughly. It might end up depending upon who utilizes this scheme, the sensitivity of their instruments, and what need they have for the massive amounts of data arising from their arrays. Simple 4 bit, 8 bit, 16 bit, 32 bit and 64 bit bit-depths would be easy enough standards to implement, though quite probably exceeding the sensitivity of the employed detectors. For most purposes, a 1 bit "Go-NoGo" sensor will probably suffice, since (1) combinatorial equations will probably be most useful, (2) the destructive use of the chemicals requires a constant refreshing of these mini-sensors (though some researchers are looking into renewable, re-usable, reset-able chemicals), and (3) Whoops . train of thpught? stream of consciousness? Caffeine epiphany? lost ....

Finally, the actual nature of the scientific literature has bearing here. For instance, some researchers have placed Odorant Binding Proteins (OBPs) of bovine and porcine origin into a SAW array and used it to sense the chemical carvone and others. Additionally, knowing the human genome, researchers might want to (1) utilize Human OBPs or (2) if possible, utilize human ORGs/ORPs (though natural extraction is probably dubious or impossible) or (3) manufacture human ORPs (or OBPs) in a lab utlizing the human genome code, as discovered and widely published. I am wholly unqualified to suggest or deny any of these courses of action. But I feel it is more likely that analogous chemicals--analog "locks" for nature's "ligand keys"--will be employed. This is a monumental task, since we do not yet know the function for more than a fraction of the approx. 400-450 ORGs/ORPs. About 10% have purported / reported function assigned to a chemical agonist. So manufacturing mimic chemcicals for chemicals whose function is not yet known, might appear, at first glance, ludicrous. And, considering the POSSIBLE uses of an olfactory bitmap, we must envision that genomics and proteomics reserchers will want to reference the Olfactory Bitmap "plan" and help modify, as their research brings more light to the subject. if "Franken-sensors" like the bovine-porcine sensor mentioned above seem a profitable avenue of exploration, the "Info Header" section should have a toggle or switch indicating that biosensors of natural or artificial orgin are being employed. { Furthermore, if standard searches for 'analogous-receptors-to-confirmed-natural-function" type sensors are being employed, likewise, a type should be specificed in the "Info Header" (revise this srntrnce!? meaning unclear?! Jan 25, 2024) }

The website OlfactionBase might help in this area, especially as more information becomes available. https://olfab.iiita.ac.in/olfactionbase/

Finally, instead of a "bottom up" approach, we might try "top down", namely, utilization of Fenaroli's Handbook of Flavor Ingredients (1968, Italian language, 1st ed. 1971, and 2nd. ed. 1975, to present, 6th edition), with a similar lookup table constructed, and perhaps, specified within the "Info Header" that Top Down is being used instead of "Bottom Up", for the code points "look up table", research could proceed from the agonist end of the research community. This researcher had collected some other reference works on the subject of aromas, fragrances and tastes, in a perfectly avoidable tragedy, has been removed from ready access to these references. It is unfortunate. I believe one of the resources, or several, were Sigma-Aldrich handbooks and brochures for chemicals they utilize in servicing the food industry. Perhaps Alfa Aesar and other chemical catalogs can properly cross reference the CAS numbers,  but I desire the actual literature, as a handy compendium and reference.

(removed extraneous, cranky comments; Jan. 25, 2024)


NEW SECTION : (UPDATE "2.0")

These notes will be somewhat random additions.

This guy nebtiobed sone high level language descriptions, and has a germ if an idea, but I jave not seen that ge took it any further with DSLs or woth SQLs. High level languages appear to nw yo be the arong direction anyway. Trabsxucers and hexadecimal and bibary and Unicide are best, in my opinion. http://www.bailis.org/blog/nosesql-and-sensedb-new-paradigms-for-crowdsourced-databases/

My own research has come pretty far and I can only mention some of it in "passing", inchoate phrases, as I have much work to do, still. I will try to update to a README 3.0 early in this year, with diagrams and meticulous, specific references.

The basic transducer should have a microfluidic, wafer scale, hermetically sealed cap. There should be PDMS quake valves and / or tesla microvalves (valvular conduits) to act as "olfactory shutters". Gustatory transducers should be, largely, a micrifluidic in nature. A so-called "Lab on a Chip".

Aromadb will be useful, as will Superscent and a few other websites whose names I forget at the moment. Watch for update README 3# Olfactory-Bitmap
Anticipatory File Structure for Massively Parallel, Anthropoid SAW and / or FOIC Chemical Sensors

The idea is not to map intensity values, collected in analog and converted and stored digitally, and store the respective spatial orientations of each of those scalar quantities, and thusly reproduce the percieved optical gamut, in a mosaic pattern called an "image", of the average H. sapiens (as is standard in the normal image bitmap with which you are presently familiar) but instead, for olfaction, a "look-up" table, similar to the image bitmap "color table" (for BMPs less than 24-bit RGB), is considered. This could be empolyed in a "Go-NoGo" sensor, binary, or in a sensor that is calibrated with separate sensitivities and can store with significant bit depth, according to each specific technology, or ligands, employed.

The "gamma-ut" (see "chanson' and the "monks of old"), it is presumed, will closely follow recent genomic and proteomic work on the Olfactory Receptor Genes (ORGs) and Olfactory Receptor Proteins (ORPs) of the "average human being". Despite the fact that science prersently lacks a viable or uncontested theory to explain the gamut of this sensory modality, it would behoove researchers to organize their research activities along certain lines of enquiry. LadnarofLadner (Randall Meyer) has identified at least two promising technologies, though more certainly exist. The first, probably better studied, would be the Surface Acoustic Wave (SAW) Chemical Sensor (See White and Voltmer, 1965; Wohltjen, 1979). Chief difficulties in implmenting this technology, as with all candidate technologies, will be finding appropriate ligands to mimic the approximately 400-450 ORGs/ORPs employed by the average H. sapiens. Furthermore, olfaction is a complicated process in biological organisms, involving slime and mucus and various other unecessary complications for electronic sensor transducer design. In particular, Olfactory Binding Proteins (OBPs) in the olfactory mucosa serve the initial role of attaching to the generally hydrophobic aroma molecules and bringing them into the aqueous environment that the olfactory neurons inhabit. It is then, here, that the ORGs/ORPs can act upon the aroma molecules.

This present project has, thus far, produced a 'lookup table" based upon the Yale/ TexasTech ORDB gene numbering scheme, and, in general, based upon synonymous entries (namely, Consensus Label, HORDE, Senomyx, EBI, ORDB, and Nei Lab.) provided in the table ( here : https://senselab.med.yale.edu/OrDB/info/humanorseqanal ), on their website, version 6 ( see here : https://senselab.med.yale.edu/OrDB/info/ordb_versions ), as download by LadnarofLadner circa February/ March of 2023. In particular, separate tables for the Consensus Label and for the ORDB "ORL #'s" have been devised, but tables for the other names will follow, perfunctorily, at my leisure. The tables will map the genes, proteins, and putative gene positions, and where available, implied functions of said ORGs/ORPs, to several private areas, and restricted areas, of the Unicode Planes, utilizing approximately 1800 to 2000 unique code points which will stand for ANALOGOUS receptor molecules, as utilized by SAW and FOIC transducer researchers who wish to embark upon a program of Massively Parallel Anthropoid Olfactory and Gustatory Transducers (MPAOGT : SAW or FOIC or "other"). It is not lost on the author that this usage of unicode is novel, but perhaps not without precedent, as "emoticons" are now covered by the Unicode. The truly novel element here is that the code points claimed here will not have any glyph assigned to them, but instead will represent an ineffable and presently indescribable, heretofore undigitizeable quanitity or quality; namely, aroma. The code points claimed will probably span the region of approx. U+ E000 to about E7FF, and in the two higher planes, 15 and 16, U+ EE000 to approx. EE7FF, and also U+ 10E000 to approx. 10E7FF. As for the restricted space I intend to claim, I think it will probably be U+ 3E000 to approx. 3E7FF. The similarity of the regions utilized will allow for efficeint and conveneint alteration of data, should it be required to change from a file utilizing one standard, into a file utilizing another standard. Its should also be mentioned that there is no patent or copyright on hexadecimmal codes, so, they are presumably free to use in any way, by any person. It just so happens that Unicode is a convenient and widely adhered to standard that happens to have nearly innumerable available code points to play with. 

So, in summary of this founding "README" file (and now README "2.0"; added Jan. 2024) I will remind you of the Image Bitmap sections: (1) Header, (2) Info Header, (3) Color Table, (4) Pixel Data. In analogy, the "BM" ASCII text indicating to application programs that the file is indeed a bitmap image, will instead be "BOBM" standing for, jocularly, "B.O. Bitmap" or, in a more serious vein, "Basic Olfactory Bit Map", in ASCII, and programs designed to access Olfactory Bitmaps should look for this to be prominently placed, in the first position of the Header, at Byte one ( and two?).

Presently, I have no analogy for the Info Header, but presumably, such a standard will evolve in the coming months.

Presently, the "lookup table", unicode code points are stationary and ordered, being, with some degree of imprecision and subject to some revision, more or less a stable pattern existing in nature ( i.e. the order of the discovered genes within the human genome determines the proper position of the analogous receptor-chemicals--and their resulting data---as designed and utilized by SAW and FOIC and other reserchers.), should not change over time, and need not be included within the file. It can be assumed; both the order of the ORGs and ORPs and of their analogs on the electronic transducers and / or such data as collected by said tranducers. This takes care of the analogy to the "Color Table", as this section is obviated, and unecessary.

Finally, the "pixel data"--or rather, "olfixel" data--can be as little as 1 bit per "code-point-address" (for a "Go-NoGo" sensor), but if various researchers are taking magnitude data and wish to keep it in the array, this "Pixel Data" section of the Image Bitmap will have its analog in the "Olfixel Data" section of an Olfactory Bitmap. Perhaps, as Section 2 , the analog to the "Info Header' takes shape, more details regarding how to store magnitude data and how to store "file type" data ( i.e. for Go-NoGo, binary type sensors, vs a file type which wishes to store magnitude data), will be developed. Today is, quite accidentally, April Fool's Day 2023 ("2.0" typo fix and revision and extension ; January 25, 2024). I think, in the latter case, magnitudes can be stored using code points of the analog chemicals as "addresses", and stored at that address a magnitude of any (reasonable) pre-determined size, specified in the "Info Header" of an Olfactory Bitmap.

But I havent really thought it through very thoroughly. It might end up depending upon who utilizes this scheme, the sensitivity of their instruments, and what need they have for the massive amounts of data arising from their arrays. Simple 4 bit, 8 bit, 16 bit, 32 bit and 64 bit bit-depths would be easy enough standards to implement, though quite probably exceeding the sensitivity of the employed detectors. For most purposes, a 1 bit "Go-NoGo" sensor will probably suffice, since (1) combinatorial equations will probably be most useful, (2) the destructive use of the chemicals requires a constant refreshing of these mini-sensors (though some researchers are looking into renewable, re-usable, reset-able chemicals), and (3) Whoops . train of thpught? stream of consciousness? Caffeine epiphany? lost ....

Finally, the actual nature of the scientific literature has bearing here. For instance, some researchers have placed Odorant Binding Proteins (OBPs) of bovine and porcine origin into a SAW array and used it to sense the chemical carvone and others. Additionally, knowing the human genome, researchers might want to (1) utilize Human OBPs or (2) if possible, utilize human ORGs/ORPs (though natural extraction is probably dubious or impossible) or (3) manufacture human ORPs (or OBPs) in a lab utlizing the human genome code, as discovered and widely published. I am wholly unqualified to suggest or deny any of these courses of action. But I feel it is more likely that analogous chemicals--analog "locks" for nature's "ligand keys"--will be employed. This is a monumental task, since we do not yet know the function for more than a fraction of the approx. 400-450 ORGs/ORPs. About 10% have purported / reported function assigned to a chemical agonist. So manufacturing mimic chemcicals for chemicals whose function is not yet known, might appear, at first glance, ludicrous. And, considering the POSSIBLE uses of an olfactory bitmap, we must envision that genomics and proteomics reserchers will want to reference the Olfactory Bitmap "plan" and help modify, as their research brings more light to the subject. if "Franken-sensors" like the bovine-porcine sensor mentioned above seem a profitable avenue of exploration, the "Info Header" section should have a toggle or switch indicating that biosensors of natural or artificial orgin are being employed. { Furthermore, if standard searches for 'analogous-receptors-to-confirmed-natural-function" type sensors are being employed, likewise, a type should be specificed in the "Info Header" (revise this srntrnce!? meaning unclear?! Jan 25, 2024) }

The website OlfactionBase might help in this area, especially as more information becomes available. https://olfab.iiita.ac.in/olfactionbase/

Finally, instead of a "bottom up" approach, we might try "top down", namely, utilization of Fenaroli's Handbook of Flavor Ingredients (1968, Italian language, 1st ed. 1971, and 2nd. ed. 1975, to present, 6th edition), with a similar lookup table constructed, and perhaps, specified within the "Info Header" that Top Down is being used instead of "Bottom Up", for the code points "look up table", research could proceed from the agonist end of the research community. This researcher had collected some other reference works on the subject of aromas, fragrances and tastes, in a perfectly avoidable tragedy, has been removed from ready access to these references. It is unfortunate. I believe one of the resources, or several, were Sigma-Aldrich handbooks and brochures for chemicals they utilize in servicing the food industry. Perhaps Alfa Aesar and other chemical catalogs can properly cross reference the CAS numbers,  but I desire the actual literature, as a handy compendium and reference.

(removed extraneous, cranky comments; Jan. 25, 2024)


NEW SECTION : (UPDATE "2.0")

These notes will be somewhat random additions.

This guy nebtiobed sone high level language descriptions, and has a germ if an idea, but I jave not seen that ge took it any further with DSLs or woth SQLs. High level languages appear to nw yo be the arong direction anyway. Trabsxucers and hexadecimal and bibary and Unicide are best, in my opinion. http://www.bailis.org/blog/nosesql-and-sensedb-new-paradigms-for-crowdsourced-databases/

My own research has come pretty far and I can only mention some of it in "passing", inchoate phrases, as I have much work to do, still. I will try to update to a README 3.0 early in this year, with diagrams and meticulous, specific references.

The basic transducer should have a microfluidic, wafer scale, hermetically sealed cap. There should be PDMS quake valves and / or tesla microvalves (valvular conduits) to act as "olfactory shutters". Gustatory transducers should be, largely, a micrifluidic in nature. A so-called "Lab on a Chip".

Aromadb will be useful, as will Superscent and a few other websites whose names I for# Olfactory-Bitmap
Anticipatory File Structure for Massively Parallel, Anthropoid SAW and / or FOIC Chemical Sensors

The idea is not to map intensity values, collected in analog and converted and stored digitally, and store the respective spatial orientations of each of those scalar quantities, and thusly reproduce the percieved optical gamut, in a mosaic pattern called an "image", of the average H. sapiens (as is standard in the normal image bitmap with which you are presently familiar) but instead, for olfaction, a "look-up" table, similar to the image bitmap "color table" (for BMPs less than 24-bit RGB), is considered. This could be empolyed in a "Go-NoGo" sensor, binary, or in a sensor that is calibrated with separate sensitivities and can store with significant bit depth, according to each specific technology, or ligands, employed.

The "gamma-ut" (see "chanson' and the "monks of old"), it is presumed, will closely follow recent genomic and proteomic work on the Olfactory Receptor Genes (ORGs) and Olfactory Receptor Proteins (ORPs) of the "average human being". Despite the fact that science prersently lacks a viable or uncontested theory to explain the gamut of this sensory modality, it would behoove researchers to organize their research activities along certain lines of enquiry. LadnarofLadner (Randall Meyer) has identified at least two promising technologies, though more certainly exist. The first, probably better studied, would be the Surface Acoustic Wave (SAW) Chemical Sensor (See White and Voltmer, 1965; Wohltjen, 1979). Chief difficulties in implmenting this technology, as with all candidate technologies, will be finding appropriate ligands to mimic the approximately 400-450 ORGs/ORPs employed by the average H. sapiens. Furthermore, olfaction is a complicated process in biological organisms, involving slime and mucus and various other unecessary complications for electronic sensor transducer design. In particular, Olfactory Binding Proteins (OBPs) in the olfactory mucosa serve the initial role of attaching to the generally hydrophobic aroma molecules and bringing them into the aqueous environment that the olfactory neurons inhabit. It is then, here, that the ORGs/ORPs can act upon the aroma molecules.

This present project has, thus far, produced a 'lookup table" based upon the Yale/ TexasTech ORDB gene numbering scheme, and, in general, based upon synonymous entries (namely, Consensus Label, HORDE, Senomyx, EBI, ORDB, and Nei Lab.) provided in the table ( here : https://senselab.med.yale.edu/OrDB/info/humanorseqanal ), on their website, version 6 ( see here : https://senselab.med.yale.edu/OrDB/info/ordb_versions ), as download by LadnarofLadner circa February/ March of 2023. In particular, separate tables for the Consensus Label and for the ORDB "ORL #'s" have been devised, but tables for the other names will follow, perfunctorily, at my leisure. The tables will map the genes, proteins, and putative gene positions, and where available, implied functions of said ORGs/ORPs, to several private areas, and restricted areas, of the Unicode Planes, utilizing approximately 1800 to 2000 unique code points which will stand for ANALOGOUS receptor molecules, as utilized by SAW and FOIC transducer researchers who wish to embark upon a program of Massively Parallel Anthropoid Olfactory and Gustatory Transducers (MPAOGT : SAW or FOIC or "other"). It is not lost on the author that this usage of unicode is novel, but perhaps not without precedent, as "emoticons" are now covered by the Unicode. The truly novel element here is that the code points claimed here will not have any glyph assigned to them, but instead will represent an ineffable and presently indescribable, heretofore undigitizeable quanitity or quality; namely, aroma. The code points claimed will probably span the region of approx. U+ E000 to about E7FF, and in the two higher planes, 15 and 16, U+ EE000 to approx. EE7FF, and also U+ 10E000 to approx. 10E7FF. As for the restricted space I intend to claim, I think it will probably be U+ 3E000 to approx. 3E7FF. The similarity of the regions utilized will allow for efficeint and conveneint alteration of data, should it be required to change from a file utilizing one standard, into a file utilizing another standard. Its should also be mentioned that there is no patent or copyright on hexadecimmal codes, so, they are presumably free to use in any way, by any person. It just so happens that Unicode is a convenient and widely adhered to standard that happens to have nearly innumerable available code points to play with. 

So, in summary of this founding "README" file (and now README "2.0"; added Jan. 2024) I will remind you of the Image Bitmap sections: (1) Header, (2) Info Header, (3) Color Table, (4) Pixel Data. In analogy, the "BM" ASCII text indicating to application programs that the file is indeed a bitmap image, will instead be "BOBM" standing for, jocularly, "B.O. Bitmap" or, in a more serious vein, "Basic Olfactory Bit Map", in ASCII, and programs designed to access Olfactory Bitmaps should look for this to be prominently placed, in the first position of the Header, at Byte one ( and two?).

Presently, I have no analogy for the Info Header, but presumably, such a standard will evolve in the coming months.

Presently, the "lookup table", unicode code points are stationary and ordered, being, with some degree of imprecision and subject to some revision, more or less a stable pattern existing in nature ( i.e. the order of the discovered genes within the human genome determines the proper position of the analogous receptor-chemicals--and their resulting data---as designed and utilized by SAW and FOIC and other reserchers.), should not change over time, and need not be included within the file. It can be assumed; both the order of the ORGs and ORPs and of their analogs on the electronic transducers and / or such data as collected by said tranducers. This takes care of the analogy to the "Color Table", as this section is obviated, and unecessary.

Finally, the "pixel data"--or rather, "olfixel" data--can be as little as 1 bit per "code-point-address" (for a "Go-NoGo" sensor), but if various researchers are taking magnitude data and wish to keep it in the array, this "Pixel Data" section of the Image Bitmap will have its analog in the "Olfixel Data" section of an Olfactory Bitmap. Perhaps, as Section 2 , the analog to the "Info Header' takes shape, more details regarding how to store magnitude data and how to store "file type" data ( i.e. for Go-NoGo, binary type sensors, vs a file type which wishes to store magnitude data), will be developed. Today is, quite accidentally, April Fool's Day 2023 ("2.0" typo fix and revision and extension ; January 25, 2024). I think, in the latter case, magnitudes can be stored using code points of the analog chemicals as "addresses", and stored at that address a magnitude of any (reasonable) pre-determined size, specified in the "Info Header" of an Olfactory Bitmap.

But I havent really thought it through very thoroughly. It might end up depending upon who utilizes this scheme, the sensitivity of their instruments, and what need they have for the massive amounts of data arising from their arrays. Simple 4 bit, 8 bit, 16 bit, 32 bit and 64 bit bit-depths would be easy enough standards to implement, though quite probably exceeding the sensitivity of the employed detectors. For most purposes, a 1 bit "Go-NoGo" sensor will probably suffice, since (1) combinatorial equations will probably be most useful, (2) the destructive use of the chemicals requires a constant refreshing of these mini-sensors (though some researchers are looking into renewable, re-usable, reset-able chemicals), and (3) Whoops . train of thpught? stream of consciousness? Caffeine epiphany? lost ....

Finally, the actual nature of the scientific literature has bearing here. For instance, some researchers have placed Odorant Binding Proteins (OBPs) of bovine and porcine origin into a SAW array and used it to sense the chemical carvone and others. Additionally, knowing the human genome, researchers might want to (1) utilize Human OBPs or (2) if possible, utilize human ORGs/ORPs (though natural extraction is probably dubious or impossible) or (3) manufacture human ORPs (or OBPs) in a lab utlizing the human genome code, as discovered and widely published. I am wholly unqualified to suggest or deny any of these courses of action. But I feel it is more likely that analogous chemicals--analog "locks" for nature's "ligand keys"--will be employed. This is a monumental task, since we do not yet know the function for more than a fraction of the approx. 400-450 ORGs/ORPs. About 10% have purported / reported function assigned to a chemical agonist. So manufacturing mimic chemcicals for chemicals whose function is not yet known, might appear, at first glance, ludicrous. And, considering the POSSIBLE uses of an olfactory bitmap, we must envision that genomics and proteomics reserchers will want to reference the Olfactory Bitmap "plan" and help modify, as their research brings more light to the subject. if "Franken-sensors" like the bovine-porcine sensor mentioned above seem a profitable avenue of exploration, the "Info Header" section should have a toggle or switch indicating that biosensors of natural or artificial orgin are being employed. { Furthermore, if standard searches for 'analogous-receptors-to-confirmed-natural-function" type sensors are being employed, likewise, a type should be specificed in the "Info Header" (revise this srntrnce!? meaning unclear?! Jan 25, 2024) }

The website OlfactionBase might help in this area, especially as more information becomes available. https://olfab.iiita.ac.in/olfactionbase/

Finally, instead of a "bottom up" approach, we might try "top down", namely, utilization of Fenaroli's Handbook of Flavor Ingredients (1968, Italian language, 1st ed. 1971, and 2nd. ed. 1975, to present, 6th edition), with a similar lookup table constructed, and perhaps, specified within the "Info Header" that Top Down is being used instead of "Bottom Up", for the code points "look up table", research could proceed from the agonist end of the research community. This researcher had collected some other reference works on the subject of aromas, fragrances and tastes, in a perfectly avoidable tragedy, has been removed from ready access to these references. It is unfortunate. I believe one of the resources, or several, were Sigma-Aldrich handbooks and brochures for chemicals they utilize in servicing the food industry. Perhaps Alfa Aesar and other chemical catalogs can properly cross reference the CAS numbers,  but I desire the actual literature, as a handy compendium and reference.

(removed extraneous, cranky comments; Jan. 25, 2024)


NEW SECTION : (UPDATE "2.0")

These notes will be somewhat random additions.

This guy nebtiobed sone high level language descriptions, and has a germ if an idea, but I jave not seen that ge took it any further with DSLs or woth SQLs. High level languages appear to nw yo be the arong direction anyway. Trabsxucers and hexadecimal and bibary and Unicide are best, in my opinion. http://www.bailis.org/blog/nosesql-and-sensedb-new-paradigms-for-crowdsourced-databases/

My own research has come pretty far and I can only mention some of it in "passing", inchoate phrases, as I have much work to do, still. I will try to update to a README 3.0 early in this year, with diagrams and meticulous, specific references.

The basic transducer should have a microfluidic, wafer scale, hermetically sealed cap. There should be PDMS quake valves and / or tesla microvalves (valvular conduits) to act as "olfactory shutters". Gustatory transducers should be, largely, a micrifluidic in nature. A so-called "Lab on a Chip".

Aromadb will be useful, as will Superscent and a few other websites whose names I forget at the moment. Watch for update README 3# Olfactory-Bitmap
Anticipatory File Structure for Massively Parallel, Anthropoid SAW and / or FOIC Chemical Sensors

The idea is not to map intensity values, collected in analog and converted and stored digitally, and store the respective spatial orientations of each of those scalar quantities, and thusly reproduce the percieved optical gamut, in a mosaic pattern called an "image", of the average H. sapiens (as is standard in the normal image bitmap with which you are presently familiar) but instead, for olfaction, a "look-up" table, similar to the image bitmap "color table" (for BMPs less than 24-bit RGB), is considered. This could be empolyed in a "Go-NoGo" sensor, binary, or in a sensor that is calibrated with separate sensitivities and can store with significant bit depth, according to each specific technology, or ligands, employed.

The "gamma-ut" (see "chanson' and the "monks of old"), it is presumed, will closely follow recent genomic and proteomic work on the Olfactory Receptor Genes (ORGs) and Olfactory Receptor Proteins (ORPs) of the "average human being". Despite the fact that science prersently lacks a viable or uncontested theory to explain the gamut of this sensory modality, it would behoove researchers to organize their research activities along certain lines of enquiry. LadnarofLadner (Randall Meyer) has identified at least two promising technologies, though more certainly exist. The first, probably better studied, would be the Surface Acoustic Wave (SAW) Chemical Sensor (See White and Voltmer, 1965; Wohltjen, 1979). Chief difficulties in implmenting this technology, as with all candidate technologies, will be finding appropriate ligands to mimic the approximately 400-450 ORGs/ORPs employed by the average H. sapiens. Furthermore, olfaction is a complicated process in biological organisms, involving slime and mucus and various other unecessary complications for electronic sensor transducer design. In particular, Olfactory Binding Proteins (OBPs) in the olfactory mucosa serve the initial role of attaching to the generally hydrophobic aroma molecules and bringing them into the aqueous environment that the olfactory neurons inhabit. It is then, here, that the ORGs/ORPs can act upon the aroma molecules.

This present project has, thus far, produced a 'lookup table" based upon the Yale/ TexasTech ORDB gene numbering scheme, and, in general, based upon synonymous entries (namely, Consensus Label, HORDE, Senomyx, EBI, ORDB, and Nei Lab.) provided in the table ( here : https://senselab.med.yale.edu/OrDB/info/humanorseqanal ), on their website, version 6 ( see here : https://senselab.med.yale.edu/OrDB/info/ordb_versions ), as download by LadnarofLadner circa February/ March of 2023. In particular, separate tables for the Consensus Label and for the ORDB "ORL #'s" have been devised, but tables for the other names will follow, perfunctorily, at my leisure. The tables will map the genes, proteins, and putative gene positions, and where available, implied functions of said ORGs/ORPs, to several private areas, and restricted areas, of the Unicode Planes, utilizing approximately 1800 to 2000 unique code points which will stand for ANALOGOUS receptor molecules, as utilized by SAW and FOIC transducer researchers who wish to embark upon a program of Massively Parallel Anthropoid Olfactory and Gustatory Transducers (MPAOGT : SAW or FOIC or "other"). It is not lost on the author that this usage of unicode is novel, but perhaps not without precedent, as "emoticons" are now covered by the Unicode. The truly novel element here is that the code points claimed here will not have any glyph assigned to them, but instead will represent an ineffable and presently indescribable, heretofore undigitizeable quanitity or quality; namely, aroma. The code points claimed will probably span the region of approx. U+ E000 to about E7FF, and in the two higher planes, 15 and 16, U+ EE000 to approx. EE7FF, and also U+ 10E000 to approx. 10E7FF. As for the restricted space I intend to claim, I think it will probably be U+ 3E000 to approx. 3E7FF. The similarity of the regions utilized will allow for efficeint and conveneint alteration of data, should it be required to change from a file utilizing one standard, into a file utilizing another standard. Its should also be mentioned that there is no patent or copyright on hexadecimmal codes, so, they are presumably free to use in any way, by any person. It just so happens that Unicode is a convenient and widely adhered to standard that happens to have nearly innumerable available code points to play with. 

So, in summary of this founding "README" file (and now README "2.0"; added Jan. 2024) I will remind you of the Image Bitmap sections: (1) Header, (2) Info Header, (3) Color Table, (4) Pixel Data. In analogy, the "BM" ASCII text indicating to application programs that the file is indeed a bitmap image, will instead be "BOBM" standing for, jocularly, "B.O. Bitmap" or, in a more serious vein, "Basic Olfactory Bit Map", in ASCII, and programs designed to access Olfactory Bitmaps should look for this to be prominently placed, in the first position of the Header, at Byte one ( and two?).

Presently, I have no analogy for the Info Header, but presumably, such a standard will evolve in the coming months.

Presently, the "lookup table", unicode code points are stationary and ordered, being, with some degree of imprecision and subject to some revision, more or less a stable pattern existing in nature ( i.e. the order of the discovered genes within the human genome determines the proper position of the analogous receptor-chemicals--and their resulting data---as designed and utilized by SAW and FOIC and other reserchers.), should not change over time, and need not be included within the file. It can be assumed; both the order of the ORGs and ORPs and of their analogs on the electronic transducers and / or such data as collected by said tranducers. This takes care of the analogy to the "Color Table", as this section is obviated, and unecessary.

Finally, the "pixel data"--or rather, "olfixel" data--can be as little as 1 bit per "code-point-address" (for a "Go-NoGo" sensor), but if various researchers are taking magnitude data and wish to keep it in the array, this "Pixel Data" section of the Image Bitmap will have its analog in the "Olfixel Data" section of an Olfactory Bitmap. Perhaps, as Section 2 , the analog to the "Info Header' takes shape, more details regarding how to store magnitude data and how to store "file type" data ( i.e. for Go-NoGo, binary type sensors, vs a file type which wishes to store magnitude data), will be developed. Today is, quite accidentally, April Fool's Day 2023 ("2.0" typo fix and revision and extension ; January 25, 2024). I think, in the latter case, magnitudes can be stored using code points of the analog chemicals as "addresses", and stored at that address a magnitude of any (reasonable) pre-determined size, specified in the "Info Header" of an Olfactory Bitmap.

But I havent really thought it through very thoroughly. It might end up depending upon who utilizes this scheme, the sensitivity of their instruments, and what need they have for the massive amounts of data arising from their arrays. Simple 4 bit, 8 bit, 16 bit, 32 bit and 64 bit bit-depths would be easy enough standards to implement, though quite probably exceeding the sensitivity of the employed detectors. For most purposes, a 1 bit "Go-NoGo" sensor will probably suffice, since (1) combinatorial equations will probably be most useful, (2) the destructive use of the chemicals requires a constant refreshing of these mini-sensors (though some researchers are looking into renewable, re-usable, reset-able chemicals), and (3) Whoops . train of thpught? stream of consciousness? Caffeine epiphany? lost ....

Finally, the actual nature of the scientific literature has bearing here. For instance, some researchers have placed Odorant Binding Proteins (OBPs) of bovine and porcine origin into a SAW array and used it to sense the chemical carvone and others. Additionally, knowing the human genome, researchers might want to (1) utilize Human OBPs or (2) if possible, utilize human ORGs/ORPs (though natural extraction is probably dubious or impossible) or (3) manufacture human ORPs (or OBPs) in a lab utlizing the human genome code, as discovered and widely published. I am wholly unqualified to suggest or deny any of these courses of action. But I feel it is more likely that analogous chemicals--analog "locks" for nature's "ligand keys"--will be employed. This is a monumental task, since we do not yet know the function for more than a fraction of the approx. 400-450 ORGs/ORPs. About 10% have purported / reported function assigned to a chemical agonist. So manufacturing mimic chemcicals for chemicals whose function is not yet known, might appear, at first glance, ludicrous. And, considering the POSSIBLE uses of an olfactory bitmap, we must envision that genomics and proteomics reserchers will want to reference the Olfactory Bitmap "plan" and help modify, as their research brings more light to the subject. if "Franken-sensors" like the bovine-porcine sensor mentioned above seem a profitable avenue of exploration, the "Info Header" section should have a toggle or switch indicating that biosensors of natural or artificial orgin are being employed. { Furthermore, if standard searches for 'analogous-receptors-to-confirmed-natural-function" type sensors are being employed, likewise, a type should be specificed in the "Info Header" (revise this srntrnce!? meaning unclear?! Jan 25, 2024) }

The website OlfactionBase might help in this area, especially as more information becomes available. https://olfab.iiita.ac.in/olfactionbase/

Finally, instead of a "bottom up" approach, we might try "top down", namely, utilization of Fenaroli's Handbook of Flavor Ingredients (1968, Italian language, 1st ed. 1971, and 2nd. ed. 1975, to present, 6th edition), with a similar lookup table constructed, and perhaps, specified within the "Info Header" that Top Down is being used instead of "Bottom Up", for the code points "look up table", research could proceed from the agonist end of the research community. This researcher had collected some other reference works on the subject of aromas, fragrances and tastes, in a perfectly avoidable tragedy, has been removed from ready access to these references. It is unfortunate. I believe one of the resources, or several, were Sigma-Aldrich handbooks and brochures for chemicals they utilize in servicing the food industry. Perhaps Alfa Aesar and other chemical catalogs can properly cross reference the CAS numbers,  but I desire the actual literature, as a handy compendium and reference.

(removed extraneous, cranky comments; Jan. 25, 2024)


NEW SECTION : (UPDATE "2.0")

These notes will be somewhat random additions.

This guy nebtiobed sone high level language descriptions, and has a germ if an idea, but I jave not seen that ge took it any further with DSLs or woth SQLs. High level languages appear to nw yo be the arong direction anyway. Trabsxucers and hexadecimal and bibary and Unicide are best, in my opinion. http://www.bailis.org/blog/nosesql-and-sensedb-new-paradigms-for-crowdsourced-databases/

My own research has come pretty far and I can only mention some of it in "passing", inchoate phrases, as I have much work to do, still. I will try to update to a README 3.0 early in this year, with diagrams and meticulous, specific references.

The basic transducer should have a microfluidic, wafer scale, hermetically sealed cap. There should be PDMS quake valves and / or tesla microvalves (valvular conduits) to act as "olfactory shutters". Gustatory transducers should be, largely, a micrifluidic in nature. A so-called "Lab on a Chip".

Aromadb will be useful, as will Superscent and a few other websites whose names I forget at the moment. Watch for update README 3.0.


.0.


get at the moment. Watch for update README 3.0.


.0.


